### PR TITLE
add client for handling CRUD operations for rate limits

### PR DESCRIPTION
**Pull request by nleeper**
Making this PR first, then move on to refactor after approval.

### CODE REVIEW

File: dto/src/main/scala/com/dwolla/cloudflare/domain/dto/ratelimits/RateLimitMatchDTO.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+case class RateLimitMatchRequestDTO (
+  methods: Option[List[String]],
+  schemes: Option[List[String]],
+  url: String
+) extends JsonWritable
+
+case class RateLimitMatchResponseDTO (
+  status: Option[List[Int]],
+  origin_traffic: Option[Boolean],
```
Comment (coreyjonoliver) [inline dto/src/main/scala/com/dwolla/cloudflare/domain/dto/ratelimits/RateLimitMatchDTO.scala]: I didn't see this field listed in the Cloudflare API docs: https://api.cloudflare.com/#rate-limits-for-a-zone-rate-limit-details. Out of curiosity, is this something undocumented you encountered?

File: dto/src/main/scala/com/dwolla/cloudflare/domain/dto/ratelimits/RateLimitMatchDTO.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+) extends JsonWritable
+
+case class RateLimitMatchRequestDTO (
+  methods: Option[List[String]],
+  schemes: Option[List[String]],
+  url: String
+) extends JsonWritable
+
+case class RateLimitMatchResponseDTO (
+  status: Option[List[Int]],
```
Comment (coreyjonoliver) [inline dto/src/main/scala/com/dwolla/cloudflare/domain/dto/ratelimits/RateLimitMatchDTO.scala]: Could usages of `List[]` be converted to `Seq[]`? `List` imposes several implementation details that aren't necessarily required when a more abstract type would do.

File: dto/src/main/scala/com/dwolla/cloudflare/domain/dto/ratelimits/RateLimitMatchDTO.scala (hunk 1 above)
Comment (nleeper) [reply dto/src/main/scala/com/dwolla/cloudflare/domain/dto/ratelimits/RateLimitMatchDTO.scala]: Yeah, I've been using the fields documented here: https://api.cloudflare.com/#rate-limits-for-a-zone-properties

They seem to be more complete on what actually *could* be returned from the API.

File: dto/src/main/scala/com/dwolla/cloudflare/domain/dto/ratelimits/RateLimitMatchDTO.scala (hunk 2 above)
Comment (nleeper) [reply dto/src/main/scala/com/dwolla/cloudflare/domain/dto/ratelimits/RateLimitMatchDTO.scala]: Good call. Updated places where List is not needed to Seq.

**Pull request merged**